src/EditorFeatures/CSharpTest/CodeActions/ConvertToInterpolatedString/ConvertToInterpolatedStringTests.cs

```diff
@@ -0,0 +1,528 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertToInterpolatedString;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeActions.ConvertToInterpolatedString
+{
+    public class ConvertToInterpolatedStringTests : AbstractCSharpCodeActionTest
+    {
+        protected override object CreateCodeRefactoringProvider(Workspace workspace) =>
+            new ConvertToInterpolatedStringRefactoringProvider();
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertToInterpolatedString)]
+        public async Task TestSingleItemSubstitution()
+        {
+            await TestAsync(
+@"using System;
+class T
+{
+    void M()
+    {
+        var a = [|string.Format(""{0}"", 1)|];
+    }
+    }",
+@"using System;
+class T
+{
+    void M()
+    {
+        var a = $""{1}"";
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertToInterpolatedString)]
+        public async Task TestItemOrdering()
+        {
+            await TestAsync(
+@"using System;
+class T
+{
+    void M()
+    {
+        var a = [|string.Format(""{0}{1}{2}"", 1, 2, 3)|];
+    }
+}",
+@"using System;
+class T
+{
+    void M()
+    {
+        var a = $""{1}{2}{3}"";
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertToInterpolatedString)]
+        public async Task TestItemOrdering2()
+        {
+            await TestAsync(
+@"using System;
+class T
+{
+    void M()
+    {
+        var a = [|string.Format(""{0}{2}{1}"", 1, 2, 3)|];
+    }
+}",
+@"using System;
+class T
+{
+    void M()
+    {
+        var a = $""{1}{3}{2}"";
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertToInterpolatedString)]
+        public async Task TestItemOrdering3()
+        {
+            await TestAsync(
+@"using System;
+class T
+{
+    void M()
+    {
+        var a = [|string.Format(""{0}{0}{0}"", 1, 2, 3)|];
+    }
+}",
+@"using System;
+class T
+{
+    void M()
+    {
+        var a = $""{1}{1}{1}"";
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertToInterpolatedString)]
+        public async Task TestItemOutsideRange()
```

Comment:
I'm not following what you guys are talking about here. The test range below seems reasonable to me.